mm2src/coins/utxo/utxo_withdraw.rs

```diff
@@ -469,12 +456,11 @@ where
                     .clone();
                 let my_address = AddressBuilder::new(
                     addr_format,
-                    AddressHashEnum::AddressHash(key_pair.public().address_hash()),
                     coin.as_ref().conf.checksum_type,
                     coin.as_ref().conf.address_prefixes.clone(),
                     coin.as_ref().conf.bech32_hrp.clone(),
                 )
-                .as_pkh()
+                .as_pkh_from_pk(*key_pair.public())
```

Comment:
I guess here we assume that HD accounts can be P2PK outputs too. So cases for list unspent and get balance should be covered

Reply:
Ummm, I didn't give that a thought until now tbh. Addresses in HD accounts could definitely have some p2pk utxos but very unlikely (cc @shamardy).

We could support that in list_unspent_group and display_balancess, but if it's the case that they never have p2pk outputs then it would be a unnecessary overhead to query for p2pk balance of each of the HD account addresses every time.

Reply:
> Ummm, I didn't give that a thought until now tbh. Addresses in HD accounts could definitely have some p2pk utxos but very unlikely

Let's not support p2pk for HD wallet, I don't think anyone using HD wallet will have p2pk utxos, also BIP44 was for p2pkh and BIP84 was for p2wpkh, there was no mention of supporting p2pk in any of the HD wallet related BIPs as far as I know.